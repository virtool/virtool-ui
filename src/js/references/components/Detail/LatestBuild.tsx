import { BoxGroupSection, Link, NoneFoundSection, RelativeTime } from "@base";
import { ReferenceBuild } from "@references/types";
import React from "react";
import styled from "styled-components";

const StyledLatestBuild = styled(BoxGroupSection)`
    align-items: center;
    display: flex;

    a {
        margin-left: auto;
    }
`;

type LatestBuildProps = {
    id: string;
    /** Information related to the latest index build */
    latestBuild: ReferenceBuild;
};

/**
 * Displays the latest index build information associated with the reference
 */
export function LatestBuild({ id, latestBuild }: LatestBuildProps) {
    if (latestBuild) {
        return (
            <StyledLatestBuild>
                <div>
                    <strong>
                        <Link to={`/refs/${id}/indexes/${latestBuild.id}`}>
                            Index {latestBuild.version}
                        </Link>
                    </strong>
                    <span>
                        &nbsp;/ Created{" "}
                        <RelativeTime time={latestBuild.created_at} /> by{" "}
                        {latestBuild.user.handle}
                    </span>
                </div>
            </StyledLatestBuild>
        );
    }

    return <NoneFoundSection noun="index builds" />;
}
