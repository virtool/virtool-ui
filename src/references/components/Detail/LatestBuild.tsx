import React from "react";
import styled from "styled-components";
import BoxGroupSection from "../../../base/BoxGroupSection";
import Link from "../../../base/Link";
import NoneFoundSection from "../../../base/NoneFoundSection";
import RelativeTime from "../../../base/RelativeTime";
import { ReferenceBuild } from "../../types";

const StyledLatestBuild = styled(BoxGroupSection)`
    align-items: center;
    display: flex;

    a {
        margin-left: auto;
    }
`;

type LatestBuildProps = {
    id: string;
    /** Information related to the latest index build */
    latestBuild: ReferenceBuild;
};

/**
 * Displays the latest index build information associated with the reference
 */
export function LatestBuild({ id, latestBuild }: LatestBuildProps) {
    if (latestBuild) {
        return (
            <StyledLatestBuild>
                <div>
                    <strong>
                        <Link to={`/refs/${id}/indexes/${latestBuild.id}`}>
                            Index {latestBuild.version}
                        </Link>
                    </strong>
                    <span>
                        &nbsp;/ Created{" "}
                        <RelativeTime time={latestBuild.created_at} /> by{" "}
                        {latestBuild.user.handle}
                    </span>
                </div>
            </StyledLatestBuild>
        );
    }

    return <NoneFoundSection noun="index builds" />;
}
