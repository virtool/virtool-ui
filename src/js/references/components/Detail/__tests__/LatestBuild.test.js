import { LatestBuild } from "../LatestBuild";

describe("<LatestBuild />", () => {
    const props = {
        id: "foo",
        latestBuild: {
            id: "bar",
            version: "1.0",
            created_at: "2",
            user: { id: "baz" }
        }
    };

    it("should render when [latestBuild!=null]", () => {
        const wrapper = shallow(<LatestBuild {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render when [latestBuild=null]", () => {
        props.latestBuild = null;
        const wrapper = shallow(<LatestBuild {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
