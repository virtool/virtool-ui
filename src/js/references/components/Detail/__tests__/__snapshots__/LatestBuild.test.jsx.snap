// Vitest Snapshot v1

exports[`<LatestBuild /> > should render when [latestBuild!=null] 1`] = `
<Styled(BoxGroupSection)>
  <div>
    <strong>
      <Link
        to="/refs/foo/indexes/bar"
      >
        Index 
        1.0
      </Link>
    </strong>
    <span>
      Â / Created 
      <RelativeTime
        time="2"
      />
       by 
      baz_handle
    </span>
  </div>
</Styled(BoxGroupSection)>
`;

exports[`<LatestBuild /> > should render when [latestBuild=null] 1`] = `
<NoneFoundSection
  noun="index builds"
/>
`;
